mcon/parse.c

```diff
@@ -285,6 +285,54 @@ static int mongo_process_option(mongo_con_manager *manager, mongo_servers *serve
 }
 
 /* Option parser helpers */
+static int parse_auth_mechanism_properties(mongo_con_manager *manager, mongo_servers *servers, char *value, char **error_message)
+{
+	char *start, *end, *colon, *prop_name, *prop_value;
+
+	/* ignore empty value */
+	if (strlen(value) == 0) {
+		return 0;
+	}
+
+	start = value;
+
+	do {
+		end = strchr(start, ',');
+		colon = strchr(start, ':');
+
+		if (!colon) {
+			int len = strlen(start) + sizeof("Error while trying to parse auth mechanism properties: No separator for ''");
+
+			*error_message = malloc(len + 1);
```

Comment:
Just noticed that every other `malloc()` in `parse.c` does the above and adds `+ 1`. Going to leave this as-is for consistency.